akka-edge-docs/src/main/paradox/guide/2-drone-location-to-cloud-service.md

```diff
@@ -0,0 +1,321 @@
+
+Java
+:  @@snip [LocationColumn.java](/samples/grpc/restaurant-drone-deliveries-service-java/src/main/java/central/drones/LocationColumn.java) { }
+
+Finally, we add configuration to make Akka Persistence R2DBC use the `LocationColumn` when writing state:
+
+Scala
+:  @@snip [persistence.conf](/samples/grpc/restaurant-drone-deliveries-service-scala/src/main/resources/persistence.conf) { #locationColumn }
+
+Java
+:  @@snip [persistence.conf](/samples/grpc/restaurant-drone-deliveries-service-java/src/main/resources/persistence.conf) { #locationColumn }
+
+
+## gRPC service
+
+To make it possible for users of our service to query the drone overviews we define a gRPC service:
+
+Scala
+:  @@snip [drone_overview_api.proto](/samples/grpc/restaurant-drone-deliveries-service-scala/src/main/protobuf/central/drones/drone_overview_api.proto) { }
+
+Java
+:  @@snip [drone_overview_api.proto](/samples/grpc/restaurant-drone-deliveries-service-java/src/main/protobuf/central/drones/drone_overview_api.proto) { }
+
+And implement the service interface Akka gRPC generates for it.
+
+The `getDroneOverview` method asks the drone entity directly about its current state. 
+
+The `getCoarseDroneLocations` is a bit more involved, querying using the `locations` column, and then using 
+the Akka Serialization infrastructure to deserialize the found drone `State` instances. Grouping the drones
+in the same coarse grained location and finally turning that map into a protobuf response message:
+
+Scala
+:  @@snip [DroneOverviewServiceImpl.scala](/samples/grpc/restaurant-drone-deliveries-service-scala/src/main/scala/central/drones/DroneOverviewServiceImpl.scala) { }
+
+Java
+:  @@snip [DroneOverviewServiceImpl.java](/samples/grpc/restaurant-drone-deliveries-service-java/src/main/java/central/drones/DroneOverviewServiceImpl.java) { }
+
+Finally, we need to start the HTTP server, making service implementation available for calls from drones:
+
+Scala
+:  @@snip [LocalDroneControlServer.scala](/samples/grpc/restaurant-drone-deliveries-service-scala/src/main/scala/central/DroneDeliveriesServer.scala) { #composeAndBind }
+
+Java
+:  @@snip [LocalDroneControlServer.java](/samples/grpc/restaurant-drone-deliveries-service-java/src/main/java/central/DroneDeliveriesServer.java) { #composeAndBind }
+
+## Running the sample
+
+The complete sample can be downloaded from GitHub, but note that it also includes the next steps of the guide:
+
+* Java: https://github.com/akka/akka-projection/tree/main/samples/grpc/restaurant-drone-deliveries-service-java
+* Scala: https://github.com/akka/akka-projection/tree/main/samples/grpc/restaurant-drone-deliveries-service-scala
+
+As this service consumes events from the service built in the previous step, start the local-drone-control service first: 
+
+@@@ div { .group-scala }
+
+To start the local-drone-control-service:
```

Comment:
why is this only in the scala group? and it's redundant with above "start the local-drone-control service first:"